Makefile

```diff
@@ -169,6 +169,12 @@ test-conformance: $(GINKGO) $(KIND) $(SSM_PLUGIN) $(KUSTOMIZE) e2e-image ## Run
 
 test-conformance-fast: ## Run clusterctl based conformance test on workload cluster (requires Docker) using a subset of the conformance suite in parallel.
 	$(MAKE) test-conformance CONFORMANCE_E2E_ARGS="-kubetest.config-file=$(KUBETEST_FAST_CONF_PATH) -kubetest.ginkgo-nodes=5 $(E2E_ARGS)"
+
+.PHONY: test-cover
+test-cover: ## Run tests with code coverage and code generate  reports
+	source ./scripts/fetch_ext_bins.sh; fetch_tools; setup_envs; go test -v -coverprofile=out/coverage.out ./... $(TEST_ARGS)
+	go tool cover -func=out/coverage.out -o out/coverage.txt
+	go tool cover -html=out/coverage.out -o out/coverage.html
```

Comment:
This needs to go to _artifacts really, otherwise there's no information for the PR

Reply:
I think there's also a test coverage format that testgrid supports. Need to check.

Reply:
Take a look at 
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1891/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R552-R556

The thing that wasn't working was getting gopherage to convert it to junit format. if there's a decent alternative, we should do that.

Reply:
Added gopherage to the ci job: https://github.com/kubernetes/test-infra/pull/21801

This is same with how CAPZ is doing, the coverage files are copied to the artifacts folder: https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-coverage/1382456816876654592/artifacts/

Reply:
great, thanks. we still need to change out to $(ARTIFACTS) i think though?